clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -605,13 +606,27 @@ private ByteBuffer getValueBytesForKey(
         if (valueManifestContainer != null) {
           valueManifestContainer.setManifest(transientRecord.getValueManifest());
         }
-        originalValue = ByteBuffer
-            .wrap(transientRecord.getValue(), transientRecord.getValueOffset(), transientRecord.getValueLen());
+        originalValue = getCurrentValueFromTransientRecord(transientRecord);
       }
     }
     return originalValue;
   }
 
+  ByteBuffer getCurrentValueFromTransientRecord(PartitionConsumptionState.TransientRecord transientRecord) {
+    ByteBuffer compressedOriginalValue =
+        ByteBuffer.wrap(transientRecord.getValue(), transientRecord.getValueOffset(), transientRecord.getValueLen());
+    try {
+      return getCompressionStrategy().isCompressionEnabled()
+          ? getCompressor().get()
+              .decompressAndPrependSchemaHeader(
```

Comment:
This function assumes that the backed array must have a schema header in the front, but `TransientRecord` doesn't have such condition check, which makes it hard to debug if this does happen in the future.
I still recommend we pass the position and length explicitly, and you can prepend the schema header in the returned ByteBuffer, which is fine.

Reply:
It is fine to let the returned ByteBuffer has schema id prepended, but it is risky to assume the backed array of input value has schema id prepended and the position of data is always 4.

Reply:
Sure, chatted offline and fixed with updated API.